src/SFA.DAS.EmploymentCheck.Functions/Application/Services/EmployerAccount/EmployerAccountService.cs

```diff
@@ -56,135 +60,87 @@ public class EmployerAccountService
+                var json = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                content = JsonConvert.DeserializeObject<TResponse>(json);
+                var resourceList = new ResourceList((IEnumerable<ResourceViewModel>)await Task.FromResult(content));
+                employerPayeSchemes = new EmployerPayeSchemes(employmentCheckBatch.AccountId, resourceList.Select(x => x.Id).ToList());
             }
-
-            // throws an exception if the IsSuccessStatusCode property is false
-            // so we check the IsSuccessStatsCode directly to avoid the exception
-            // enabling us to store the response
-            if (response.IsSuccessStatusCode == false)
+            catch (Exception e)
             {
-                // The API call returned an none successful code
-                // Log it and throw and exception to skip the rest of the processing
-                _logger.LogError(
-                    $"\n\n{thisMethodName}: response IsSuccessStatusCode returned from the Employer Accounts API is false the status code is [{response.StatusCode}]");
-
-                await StoreAccountsResponse(new AccountsResponse(
-                    apprenticeEmploymentCheck.Id,
-                    apprenticeEmploymentCheck.CorrelationId,
-                    apprenticeEmploymentCheck.AccountId,
-                    "NULL", // payeSchemes
-                    response.ToString(),
-                    (short) response.StatusCode));
-                throw
-                    new InvalidOperationException(
-                        nameof(response)); // TODO: Create a custom business exception for this condition
+                accountsResponse.HttpResponse = e.Message;
+                content = (TResponse)new object();
+                _logger.LogError($"EmployerAccountService.Get(): ERROR: {accountsResponse.HttpResponse}");
             }
-
-
-            var json = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
-            if (string.IsNullOrEmpty(json))
+            finally
             {
-                // read of content failed
-                _logger.LogError(
-                    $"\n\n{thisMethodName}: converting the response Content to string returned an Empty/Null string (the status code is [{response.StatusCode}])");
-
-                await StoreAccountsResponse(new AccountsResponse(
-                    apprenticeEmploymentCheck.Id,
-                    apprenticeEmploymentCheck.CorrelationId,
-                    apprenticeEmploymentCheck.AccountId,
-                    "NULL", // payeSchemes
-                    response.ToString(),
-                    (short) response.StatusCode));
-                throw
-                    new InvalidOperationException(
-                        nameof(response)); // TODO: Create a custom business exception for this condition
+                try
+                {
```

Comment:
a couple of warnings of possible null references here.
also try / catch inside finally, can this be avoided at all?

Reply:
Sorted, copy-and-past errors. The try/catch in the finally is to catch the exception from the _respository.Save() if it's saving a duplicate. This is only temporary until we switch to 'atomic' individual message processing where we don't need to stop exceptions bypassing the rest of the rows in a batch.

It would be relatively straight-forward to do the switch to 'atomic' processing so maybe the sprint after next we can implement the change and go back to the 'clean' version you had after you did story 84.